guacamole-common-js/src/main/webapp/modules/Parser.js

```diff
@@ -209,8 +209,14 @@ Guacamole.Parser = function Parser() {
                     // completely parsed, so that the next call to receive()
                     // need not append to the buffer unnecessarily
                     if (elementEnd + 1 === buffer.length) {
-                        elementEnd = -1;
-                        buffer = '';
+                        if (isBuffer) {
+                            startIndex = buffer.length;
+                            break;
+                        }
+                        else {
+                            elementEnd = -1;
+                            buffer = '';
+                        }
                     }
```

Comment:
Would it make more sense to just add `!isBuffer &&` to the overall condition here? I would think that the existing logic will already ensure `startIndex` is set correctly and that the loop breaks once the buffer is exhausted.

Reply:
Or perhaps I'm just not seeing under what conditions `startIndex` would be incorrectly set...

Reply:
That seems reasonable. I'll test it!

Reply:
I tested your suggestion: it was equally effective and much more concise :) The PR has been updated